### ISSUE TITLE
Please remove Dio from dependencies

### ISSUE THREAD

**Issue by adriank** [incrediblezayed/file_saver#102]
file_saver has Dio in its dependencies. I don't use Dio in my project, so this package adds unnecessary files to the app bundle. The problem is that Dio can cause security problems in the future, and including it in the project when it's not used is not a good idea. Also on Flutter web it causes the bundle to be slightly bigger for no reason.

The solution would be to switch to Http (that is probably included in every Flutter app by some other dependency) or move the downloading step to documentation. 

Either way, this would require releasing a breaking change, so for now I will clone the repository and remove Dio on my own.

**Comment by Levi-Lesches** [comment]
@incrediblezayed I agree with this, I would love to use this plugin but bundling Dio when I wasn't already planning on it is a decent change. And I was just trying to use this to save pre-written text, so Dio is _that_ much more of a waste.

**Comment by incrediblezayed** [comment]
Done in 0.5.0: `dio` is gone, replaced by `package:http`. `dioClient` became `httpClient` (an `http.Client?`), `transformDioResponse` and `LinkDetails.responseType` were removed, and 4xx/5xx responses now throw instead of writing the error body to disk. See the CHANGELOG for the migration notes.
